### PR TITLE
Update ruff to 0.7.0

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -150,7 +150,7 @@ select = [
     "RUF026", # default factory kwarg
     "TCH",
     "TRY002", # ban vanilla raise (todo fix NOQAs)
-    "TRY302",
+    "TRY203",
```
Comment (ezyang) [inline pyproject.toml:153]: What's going on here?

**Review by ezyang — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (cyyever) [reply pyproject.toml:153]: The useless-try-except rule (in our tryceratops category) has been recoded from TRY302 to
TRY203 (https://github.com/astral-sh/ruff/pull/13502). This ensures Ruff's code is consistent with
the same rule in the [tryceratops](https://github.com/guilatrova/tryceratops) linter.

**Pull request closed without merging**